tests/slithir/operation_reads_and_writes.sol

```diff
@@ -0,0 +1,422 @@
+        Placeholder destination;
+        uint msgValue;
+        uint callGas;
+        uint argument;
+        destination.callMe{value: msgValue, gas: callGas}(argument);
+    }
+    
+}
+
+contract HighLevelCallWriteAll {
+    uint retValue; 
+
+    function writeAllStateVariables() external {
+        Placeholder destination = new Placeholder();
+        retValue =  destination.callMe(0);
+    }
+
+    function writeAllLocalVariables() external {
+        Placeholder destination = new Placeholder();
+        uint retValue;
+        retValue =  destination.callMe(0);
+    }
+}
+
+contract LowLevelCallReadAll {
+    uint msgValue;
+    uint callGas;
+    address destination;
+    uint argument;
+
+    function readAllStateVariables() external {
+        destination.call{value: msgValue, gas: callGas}(abi.encodeWithSignature("callMe(uint256)", argument));
+    }
+
+    function readAllLocalVariables() external {
+        address destination;
+        uint msgValue;
+        uint callGas;
+        uint argument;
+        destination.call{value: msgValue, gas: callGas}(abi.encodeWithSignature("callMe(uint256)", argument));
+    }
+    
+}
+
+contract LowLevelCallWriteAll {
+    bool success; 
+    bytes data;
+
+    function writeAllStateVariables() external {
+        Placeholder destination = new Placeholder();
+        (success, data) = address(destination).call(abi.encodeWithSignature("callMe(uint256)", 0));
+    }
+
+    function writeAllLocalVariables() external {
+        Placeholder destination = new Placeholder();
+        (bool success, bytes memory data) = address(destination).call(abi.encodeWithSignature("callMe(uint256)", 0));
+    }
+}
+
+contract ReturnReadAll {
+    uint stateVar;
+
+    function readAllStateVariables() external returns (uint) {
+        return stateVar;
+    }
+
+    function readAllLocalVariables() external returns (uint) {
+        uint localVar;
+        return localVar;
+    }
+}
+
+contract NewStructureReadAll {
+    uint a;
+    uint b;
+
+    struct S {
+        uint x;
+        uint y;
+    }
+
+
+    function readAllStateVariables() external {
+        S({x: a, y: b});
+    }
+
+    function readAllLocalVariables() external {
+        uint c;
+        uint d;
+        S({x: c, y: d});
+    }
+}
+
+contract TypeConversionReadAll {
+    uint stateVar;
+
+    function readAllStateVariables() external {
+        uint localVar = uint8(stateVar);
+    }
+
+    function readAllLocalVariables() external {
+        uint8 localVar;
+        uint(localVar);
+    }
+}
+
+contract TypeConversionWriteAll {
+    uint stateVar;
+
+    function writeAllStateVariables() external {
+        uint localVar;
+        stateVar = uint8(localVar);
+    }
+
+    function writeAllLocalVariables() external {
+        uint8 localVar = uint8(stateVar);
+    }
+}
+
+contract IndexReadAll {
+    uint[] stateVar;
+    uint index;
+
+    function readAllStateVariables() external {
+        stateVar[index];
+    }
+
```

Comment:
I had to make the changes in https://github.com/crytic/slither/pull/1781/commits/5083c82e0736dc264173e2c2cdbe83b16af94c4b for this to pass